### ISSUE TITLE
[Bug] Incorrect formatting with `matches!` and `..` in patterns

### ISSUE THREAD

**Issue by rami3l** [andrewbaxter/genemichaels#3]
Given:

```rs
struct Foo {
    yes: bool,
    other: i32,
}

fn g(f: Foo) {
    assert!(matches!(f, Foo { yes, .. } if yes))
}
```

This code gets formatted to the following:

```rs
struct Foo {
    yes: bool,
    other: i32,
}

fn g(f: Foo) {
    assert!(matches!(f, Foo { yes: yes } if yes))
}
```

... which is wrong since it gives the following error:

```rs
error[[E0027]](https://doc.rust-lang.org/stable/error-index.html#E0027): pattern does not mention field `other`
 --> src/lib.rs:7:25
  |
7 |     assert!(matches!(f, Foo { yes: yes } if yes))
  |                         ^^^^^^^^^^^^^^^^ missing field `other`
  |
help: include the missing field in the pattern
  |
7 |     assert!(matches!(f, Foo { yes: yes, other } if yes))
  |                                       ~~~~~~~~~
help: if you don't care about this missing field, you can explicitly ignore it
  |
7 |     assert!(matches!(f, Foo { yes: yes, .. } if yes))
  |                                       ~~~~~~
```

... and the following warning after fixing the error:

```rs
warning: the `yes:` in this pattern is redundant
 --> src/lib.rs:7:31
  |
7 |     assert!(matches!(f, Foo { yes: yes, .. } if yes))
  |                               ^^^^^^^^ help: use shorthand field pattern: `yes`
  |
  = note: `#[warn(non_shorthand_field_patterns)]` on by default
```

**Comment by andrewbaxter** [comment]
Thanks for the issue! Made https://github.com/andrewbaxter/genemichaels/issues/4 for the warning

**Comment by andrewbaxter** [comment]
I think the first issue has been fixed, I'll release int in 1.1.10 after I do #4 as well, feel free to reopen if I didn't get it right.
